### PR TITLE
Itemized extrinsic

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is to update lib.rs/tests.rs/mock.rs to use stateful_tree_storage 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: pallets/stateful-storage/src/tests.rs [hunk 1]
```diff
@@ -30,143 +47,19 @@ fn upsert_page_too_large_errors() {
...
-	let bounded: BoundedVec<u8, _> = BoundedVec::try_from(buffer).unwrap();
-	let page = ItemPage::<Test>::new(1, 1u16, bounded);
-
-	// act
-	let parsed = page.parse();
-
-	// assert
-	assert_eq!(parsed, Err(PageError::ErrorParsing("wrong header size")));
-}
-
```
Comment (saraswatpuneet) [inline pallets/stateful-storage/src/tests.rs:156]: these tests are moved to types.rs

Comment (saraswatpuneet) [reply pallets/stateful-storage/src/tests.rs:156]: tests.rs would contain paller specific tests

File: pallets/stateful-storage/src/lib.rs [hunk 2]
```diff
@@ -235,14 +234,21 @@ pub mod pallet {
 
 			match updated_page.is_empty() {
 				true => {
-					ChildTreeStorage::kill(&state_owner_msa_id, storage_key);
+					StatefulChildTree::kill::<Page<T::MaxItemizedPageSizeBytes>>(
```
Comment (aramikm) [inline pallets/stateful-storage/src/lib.rs:237]: kill doesn't need to have a type. We can remove the generic type from kill function

**Review by aramikm — APPROVED** [review]
looks good! Added one improvement comment!

**Review by JoeCap08055 — APPROVED** [review]

File: pallets/stateful-storage/src/lib.rs (hunk 2 above)
Comment (saraswatpuneet) [reply pallets/stateful-storage/src/lib.rs:237]: @JoeCap08055  can include this in his branch

**Pull request merged**